File: core-java-practice/leet-code-codebase/palindromeNumber.java

import java.util.Scanner;

public class palindromeNumber {

    public static void main(String[] args) {
        //  Scanner for read input from the user
        Scanner sc = new Scanner(System.in);
        //  Read the no. of check
        int n = sc.nextInt();
        System.out.println(isPalindrome(n));
    }

    public static boolean isPalindrome(int x) {
        //  -ve no. are not plaindrome
        if (x < 0) {
            return false;
        }
        int rev = 0;
        int a = x;
        // reverse the no. 
        while (x > 0) {
            int rem = x % 10;
            rev = rev * 10 + rem;
            x /= 10;
        }
        return a == rev;
    }
}

File: core-csharp-practice/leet-code-codebase/PalindromeNumber.cs
using System;

class PalindromeNumber
{
    static void Main()
    {
        // Read input from user
        Console.WriteLine("Enter a number:");
        string input = Console.ReadLine();

        if (int.TryParse(input, out int number))
        {
            bool result = IsPalindrome(number);
            Console.WriteLine(result);
        }
        else
        {
            Console.WriteLine("Invalid input");
        }
    }

    static bool IsPalindrome(int x)
    {
        // Negative numbers are not palindrome
        if (x < 0)
        {
            return false;
        }

        int rev = 0;
        int original = x;

        // Reverse the number
        while (x > 0)
        {
            int rem = x % 10;
            rev = rev * 10 + rem;
            x = x / 10;
        }

        return original == rev;
    }
}
